sdk/include/psdk/driverspecs.h

```diff
@@ -78,63 +83,63 @@
+#define __drv_innerNeverHoldGlobal(kind,param)      _Pre_ _SA_annotes2(SAL_neverHoldGlobal, #kind, param\t)
+#define __drv_innerReleasesGlobal(kind,param)       _Post_ _SA_annotes2(SAL_releaseGlobal,#kind, param\t)
+#define __drv_interlocked                           _Interlocked_operand_
+#define __drv_inTry                                 _Pre_ _SA_annotes1(SAL_inTry,__yes)
+#define __drv_IoGetDmaAdapter                       _Kernel_IoGetDmaAdapter_
 #define __drv_isCancelIRQL                          _IRQL_is_cancel_
-#define __drv_isObjectPointer
-#define __drv_KMDF
-#define __drv_maxFunctionIRQL(irql)
-#define __drv_maxIRQL(irql)
-#define __drv_minFunctionIRQL(irql)
-#define __drv_minIRQL(irql)
-#define __drv_Mode_impl(x)
-#define __drv_mustHold(kind)
-#define __drv_mustHoldCancelSpinLock
-#define __drv_mustHoldCriticalRegion
-#define __drv_mustHoldGlobal(kind,param)
-#define __drv_mustHoldPriorityRegion
-#define __drv_NDIS
-#define __drv_neverHold(kind)
-#define __drv_neverHoldCancelSpinLock
-#define __drv_neverHoldCriticalRegion
-#define __drv_neverHoldGlobal(kind,param)
-#define __drv_neverHoldPriorityRegion
-#define __drv_nonConstant
-#define __drv_notInTry
-#define __drv_notPointer
-#define __drv_out_deref(annotes)
-#define __drv_out(annotes)
-#define __drv_preferredFunction(func,why)
-#define __drv_raisesIRQL(irql)
-#define __drv_releasesCancelSpinLock
-#define __drv_releasesCriticalRegion
-#define __drv_releasesExclusiveResource(kind)
-#define __drv_releasesExclusiveResourceGlobal(kind,param)
-#define __drv_releasesPriorityRegion
-#define __drv_releasesResource(kind)
-#define __drv_releasesResourceGlobal(kind,param)
-#define __drv_reportError(why)
-#define __drv_requiresIRQL(irql)
-#define __drv_restoresIRQL
-#define __drv_restoresIRQLGlobal(kind,param)
-#define __drv_ret(annotes)
-#define __drv_sameIRQL
-#define __drv_savesIRQL
-#define __drv_savesIRQLGlobal(kind,param)
-#define __drv_strictType(typename,mode)
-#define __drv_strictTypeMatch(mode)
-#define __drv_unit(p)
-#define __drv_useCancelIRQL
-#define __drv_valueIs(arglist)
-#define __drv_WDM
-#define __drv_when(cond,annotes)
-#define __internal_kernel_driver
-#define __kernel_code
-#define __kernel_driver
+#define __drv_isObjectPointer                       _Points_to_data_
+#define __drv_KMDF                                  __drv_Mode_impl(KMDF)
+#define __drv_maxFunctionIRQL(irql)                 _IRQL_always_function_max_(irql)
+#define __drv_maxIRQL(irql)                         _IRQL_requires_max_(irql)
+#define __drv_minFunctionIRQL(irql)                 _IRQL_always_function_min_(irql)
+#define __drv_minIRQL(irql)                         _IRQL_requires_min_(irql)
+#define __drv_Mode_impl(x)                          _Landmark_(x) __inline void __GENSYM(__SAL_dummy_)(void){}
+#define __drv_mustHold(kind)                        _When_(!_Arg_comp_(#kind,"Memory"), _Requires_lock_held_(_Curr_)) _When_(_Arg_comp_(#kind,"Memory"), _Kernel_requires_resource_held_(kind))
```

Comment:
Too long line.

Reply:
I don't think breaking lines makes it anymore readable.